app/main.py

```diff
@@ -2,24 +1,81 @@
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: int | str) -> None:
+        self.private_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> None:
```

Comment:
The return type of the `__get__` method should be `int`, not `None`, since it returns an integer value.